packages/fhir-keycloak-user-management/src/components/CreateEditUser/index.tsx

```diff
@@ -76,19 +136,45 @@ const practitionerUpdater =
       ],
     };
 
-    const serve = new FHIRServiceClass(baseUrl, practitionerResourceType);
-    let promise = () => serve.create(payload);
-    if (isEditMode) {
-      promise = () => serve.update(payload);
-    }
-    return promise()
-      .then(() => {
-        sendSuccessNotification(successMessage);
-        if (!isEditMode) history.push(`${URL_USER_CREDENTIALS}/${userId}`);
-      })
-      .catch(() => {
-        sendErrorNotification(t('Failed to update practitioner'));
-      });
+    const serve = new FHIRServiceClass<IPractitioner>(baseUrl, practitionerResourceType);
+    return (
+      serve
+        // use update (PUT) for both creating and updating practitioner resource
+        // because create (POST) does not honour a supplied resource id
+        // and overrides with a server provided one instead
+        .update(payload)
+        .then((res) => {
+          sendSuccessNotification(successMessage);
+          return res;
+        })
+        .then((res) => {
+          createEditGroupResource(
+            values.enabled ?? false,
+            userId,
+            `${values.firstName} ${values.lastName}`,
+            res.identifier?.find((identifier) => identifier.use === 'official')?.value ??
+              payload.id ??
+              '',
+            baseUrl,
+            t,
+            group?.id
+          )
+            .then(() =>
+              sendSuccessNotification(
+                t(`Group resource ${group ? 'updated' : 'created'} successfully`)
+              )
+            )
+            .catch(() =>
+              sendErrorNotification(t(`Failed to ${group ? 'update' : 'create'} group resource`))
+            );
+        })
+        .catch(() =>
+          sendErrorNotification(t(`Failed to ${isEditMode ? 'update' : 'create'} practitioner`))
+        )
```

Comment:
These strings will not be extracted correctly by our i18n string parser, also as best practice its recommended to prefer full phrases/sentences over combining them like so.